poly/src/polynomial/multivariate/mod.rs

```diff
@@ -0,0 +1,153 @@
+//! Work with sparse multivariate polynomials.
+use ark_ff::Field;
+use ark_std::{
+    cmp::Ordering,
+    fmt::{Debug, Error, Formatter},
+    hash::Hash,
+    ops::Deref,
+    vec::Vec,
+};
+
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+mod sparse;
+pub use sparse::SparsePolynomial;
+
+/// Describes the interface for a term of a multivariate polynomial.
+pub trait Term:
+    Clone + PartialOrd + Ord + PartialEq + Eq + Hash + Default + Debug + Deref + Send + Sync
+{
+    /// Create a new Term from a list of tuples of the form `(variable, power)`
+    fn new(term: Vec<(usize, usize)>) -> Self;
+
+    /// Returns the sum of all variable powers in `self`
+    fn degree(&self) -> usize;
+
+    /// Returns a list of variables in `self`
+    fn vars(&self) -> Vec<usize>;
+
+    /// Returns a list of variable powers in `self`
+    fn powers(&self) -> Vec<usize>;
+
+    /// Returns whether `self` is a constant
+    fn is_constant(&self) -> bool;
+
+    /// Evaluates `self` at the given `point` in the field.
+    fn evaluate<F: Field>(&self, point: &[F]) -> F;
+}
+
+/// Stores a single term in a multivariate polynomial. Each element
+/// is of the form `(variable, power)`.  
```

Comment:
What do you mean by "single term"?

Reply:
Single term = monomial. Would you prefer to use `Monomial` instead of `Term`? Or just 'single term' -> 'term'?

Reply:
Ahh I was just confused by seeing single term and then seeing a vec inside. I understand now. I'm fine with either term or monomial; which one do you think would be clearer?

Reply:
Gotcha. To me `Term` is clearer and cleaner in the code but I don't think it's a huge difference either way. Perhaps we can just make sure it's clear from the documentation that term=monomial?

Reply:
Yeah let's go with that. `Term` is also shorter